From c02cb8fdcef0a7764ef458a5a24ec4f5b986e442 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 16 May 2024 13:02:11 +0200
Subject: [PATCH 0826/1695] drivers: stm32_gpio: add support for stm32mp21

Add support for stm32mp21 compatibles:
- st,stm32mp215-pinctrl
- st,stm32mp215-z-pinctrl

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 10cf1abe24ab053fc622268f8891f8ceb83d3091 drivers: stm32_gpio: add support for stm32mp21]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I0dccd0917c37538163189f75eb51a9dd934ce865
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/380207
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_gpio.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/core/drivers/stm32_gpio.c b/core/drivers/stm32_gpio.c
index 53d518229..1314b2183 100644
--- a/core/drivers/stm32_gpio.c
+++ b/core/drivers/stm32_gpio.c
@@ -1413,6 +1413,21 @@ static const struct dt_device_match stm32_pinctrl_match_table[] = {
 			.secure_extended = true,
 		},
 	},
+	{
+		.compatible = "st,stm32mp215-pinctrl",
+		.compat_data = &(struct bank_compat){
+			.secure_control = true,
+			.secure_extended = true,
+		},
+	},
+	{
+		.compatible = "st,stm32mp215-z-pinctrl",
+		.compat_data = &(struct bank_compat){
+			.gpioz = true,
+			.secure_control = true,
+			.secure_extended = true,
+		},
+	},
 	{ }
 };
 
-- 
2.47.2

